service/create/pki.go

```diff
@@ -0,0 +1,119 @@
+package create
+
+import (
+	"strings"
+
+	microerror "github.com/giantswarm/microkit/error"
+
+	"github.com/giantswarm/certctl/service/pki"
+	"github.com/giantswarm/certctl/service/token"
+)
+
+const (
+	// TODO Choose a sensible value and move to ConfigMap for V1
+	CertificateAuthorityTTL = "1440h"
+)
+
+func (s *Service) checkPKIBackend(clusterID string) bool {
+	service, err := s.getPKIService()
+	if err != nil {
+		return false
+	}
+
+	// Check PKI config.
+	mounted, err := service.IsMounted(clusterID)
+	if !mounted || err != nil {
+		return false
+	}
+	caGenerated, err := service.IsCAGenerated(clusterID)
+	if !caGenerated || err != nil {
+		return false
+	}
+	roleCreated, err := service.IsRoleCreated(clusterID)
+	if !roleCreated || err != nil {
+		return false
+	}
+
+	// PKI config is valid.
+	return true
+}
+
+func (s *Service) createPKIBackend(cert CertificateSpec) error {
+	var err error
+
+	service, err := s.getPKIService()
+	if err != nil {
+		return microerror.MaskAny(err)
+	}
+
+	// Create PKI backend
+	config := pki.CreateConfig{
+		ClusterID:        cert.ClusterID,
+		CommonName:       s.getClusterCA(cert),
+		AllowedDomains:   s.getAllowedDomainsForCA(cert),
+		AllowBareDomains: cert.AllowBareDomains,
+		TTL:              CertificateAuthorityTTL, // TTL must be longer than for the issued certs.
+	}
+
+	return service.Create(config)
+}
+
+func (s *Service) checkPKIPolicy(clusterID string) bool {
+	service, err := s.getTokenService()
+	if err != nil {
+		return false
+	}
+
+	// Check if there is a PKI policy.
+	policyCreated, err := service.IsPolicyCreated(clusterID)
+	if !policyCreated || err != nil {
+		return false
+	}
+
+	// PKI policy is valid.
+	return true
+}
+
+func (s *Service) createPKIPolicy(cert CertificateSpec) error {
+	service, err := s.getTokenService()
+	if err != nil {
+		return microerror.MaskAny(err)
+	}
+
+	// Create PKI policy TODO Use latest certctl with CreateConfig object.
+	return service.CreatePolicy(cert.ClusterID)
+}
+
+// Get the common name for the Cluster CA by removing the prefix
```

Comment:
I think we should have a common base domain and several formats to derive the actual common names for the certificates from. In the cluster-service we already do something like this. See https://github.com/giantswarm/cluster-service/blob/master/kubernetes/configmap.yaml#L31. Here it is only that the base domain is copy/paste and not use in combination. This should be reworked when we get some time for boyscouting. I would say lets try to do that right here.